### ISSUE TITLE
Norwegian Meteorological Institute source

### ISSUE THREAD

**Issue by julianfairfax** [WangDaYeeeeee/GeometricWeather#399]
It would be nice if data from the Norwegian Meteorological Institute was available. Their data is open (https://www.met.no/en/free-meteorological-data) and seems to support the whole world. GNOME and KDE Weather both use it as their sole provider of weather data. It would be nice to be able to have the same source for GNOME, KDE, and Geometric Weather.

**Comment by papjul** [comment]
Is it the same as #242? Both mention the same institute.

**Comment by julianfairfax** [comment]
Seems like it is. I didn't see that because Yr is a client

**Comment by papjul** [comment]
Can you check which source would be the best? Would one or the other provider provide better privacy, better rates limit, etc? So that we keep onky one issue opened ;)

**Comment by julianfairfax** [comment]
> Can you check which source would be the best? Would one or the other provider provide better privacy, better rates limit, etc? So that we keep onky one issue opened ;)

Do you mean between the Norwegian Meteorological Institute and Yr? Yr is just a client for the Norwegian Meteorological Institute's data, so you would use the Norwegian Meteorological Institute API directly. If we're talking about any provider, I don't know what other issues are open, but the Norwegian Meteorological Institute's data is open and I believe they have good privacy: https://www.met.no/en/About-us/privacy. They're based in Europe (and are a governmental institution), so they respect European privacy laws. I would use the Norwegian Meteorological Institute over any source if I had the option in Geometric Weather, and I think it's for the reasons i mentioned that GNOME Weather (an open and privacy-oriented application) uses it.

**Comment by papjul** [comment]
Yes, I mean what are the differences between Yr AND Norwegian Meteorological institute (besides that one is the client of the other).

- Does Yr provide the same privacy level as the other?
- Does Yr provides better free rates limit than the other?
- Does Yr aggregates other data than just Norwegian Meteorological institute (such as air pollution that Norwegian Meteorological institute might not provide, for example)?
- And other stuff you might find worth mentioning as differences.

**Comment by julianfairfax** [comment]
> Yes, I mean what are the differences between Yr AND Norwegian Meteorological institute (besides that one is the client of the other).
> 
>     * Does Yr provide the same privacy level as the other?
> 
>     * Does Yr provides better free rates limit than the other?
> 
>     * Does Yr aggregates other data than just Norwegian Meteorological institute (such as air pollution that Norwegian Meteorological institute might not provide, for example)?
> 
>     * And other stuff you might find worth mentioning as differences.

To be honest, I don't know. But all of Yr's data is just met.no's data. They simply don't have their own client, so Yr is effectively it (was a partnership). You'd have to check https://www.met.no/en/free-meteorological-data and https://hjelp.yr.no/hc/en-us/categories/200450271-About-Yr-the-API-and-our-privacy-policy I think

**Comment by julianfairfax** [comment]
See also https://api.met.no/ and https://developer.yr.no/doc/GettingStarted/. I don't believe there is a privacy or rate difference between the two. And I think all of Yr's data is simply from met.no

**Comment by julianfairfax** [comment]
> See also https://api.met.no/ and https://developer.yr.no/doc/GettingStarted/. I don't believe there is a privacy or rate difference between the two. And I think all of Yr's data is simply from met.no

https://developer.yr.no/doc/GettingStarted/ uses api.met.no anyway. There is no Yr API as far as I'm aware.

**Comment by julianfairfax** [comment]
Tl;dr: yr.no is met.no, met.no should be added because: open data, privacy laws, desktop consistency

**Comment by julianfairfax** [comment]
Is there anything I can do to help get this added faster?

**Comment by papjul** [comment]
You can have a look at how I did it for OWM: https://github.com/WangDaYeeeeee/GeometricWeather/commit/753ea5cafb9e28bb45d3ae429c1023221312b7e9

A lot of things changed since then, notably the ability to change API key, and some refactoring in Kotlin. Feel free to ask if you struggle at one point.

**Comment by julianfairfax** [comment]
> You can have a look at how I did it for OWM: [753ea5c](https://github.com/WangDaYeeeeee/GeometricWeather/commit/753ea5cafb9e28bb45d3ae429c1023221312b7e9)
> 
> A lot of things changed since then, notably the ability to change API key, and some refactoring in Kotlin. Feel free to ask if you struggle at one point.

I don't think I'm able to do that. I'm not an Android developer.

**Comment by papjul** [comment]
I started implementing MET Norway on this branch: https://github.com/papjul/GeometricWeather/tree/met
(not working yet, but I would say I only miss the daily data)

However, so far, this provider has been rather disappointing:
- They don't provide location search API, which means we need to use a second service (opted for Nominatim here)
- If you don't live in Norway and sometimes in Nordic area, you won't get current conditions, alerts, air quality. I don't think I will implement these features, but feel free to do it.
- They only provide hourly data, which means I need to implement the daily part myself from the hourly data.

**Comment by papjul** [comment]
MET Norway is implemented in #440, feel free to try the debug APK and give feedback ;)

**Comment by julianfairfax** [comment]
Completed once MR is merged, support exists in fork.
